doc/references/ImplementationOfVHDL.rst

```diff
@@ -142,21 +136,77 @@ You can either use a default clock like this:
 
 .. code-block:: VHDL
 
-    -- psl default clock is rising_edge (CLK);
-    -- psl assert always
-    --   a -> eventually! b;
+    default clock is rising_edge (CLK);
+    assert always
+      a -> eventually! b;
 
 or use a clocked expression (note the use of parentheses):
 
 .. code-block:: VHDL
 
-    -- psl assert (always a -> next[3](b)) @rising_edge (clk);
+    assert (always a -> next[3](b)) @rising_edge(clk);
 
 
 Of course only the simple subset of PSL is allowed.
 
 Currently the built-in functions are not implemented.
 
+PSL usage
+-----------------
+
+PSL annotations embedded in comments
+  GHDL understands embedded PSL annotations in VHDL files:
+
+  .. code-block:: VHDL
+
+        -- psl default clock is rising_edge (CLK);
+        -- psl assert always
+        --   a -> eventually! b;
+      end architecture rtl;
+
+  * As PSL annotations are embedded within comments, you must analyze and elaborate
+    your design with option *-fpsl* to enable PSL annotations.
+
+  * A PSL assertion statement must appear within a comment that starts
+    with the `psl` keyword. The keyword must be followed (on the
+    same line) by a PSL keyword such as `assert` or `default`.
+    To continue a PSL statement on the next line, just start a new comment.
+
+PSL annotations (VHDL-2008 only)
+  Since VHDL-2008 PSL is integrated in the VHDL language. You can use
+  PSL in a VHDL(-2008) design without embedding it in comments.
+
+  .. code-block:: VHDL
+
+        default clock is rising_edge (CLK);
+        assert always
+          a -> eventually! b;
+      end architecture rtl;
+
+  * You have to use the ``--std=08`` option.
+
+PSL vunit files
+  GHDL supports vunit (Verification Unit) files.
```

Comment:
Since we don't have an 'examples' section, might be good to add a reference to 
`testsuite/synth/psl02/`.

Reply:
I'm thinking about integrating some typical PSL use cases in the documentation, but don't know where to put them. I fear that the PSL implementation section would be too long if we put the examples there. Maybe you have an idea?

Reply:
Related to #915, I think we should have an 'examples' subdir (in the root, not a subdir of doc):

- `examples/`
  - `quick_start/`
    - `README.rst`
    - `hello/`
      - `README.rst`
      - `run.sh`
      - `hello.vhd`
    - `heartbeat/`
      - `README.rst`
      - `run.sh`
      - `heartbeat.vhd`
    - `adder/`
      - `README.rst`
      - `run.sh`
      - `adder.vhd`
      - `adder_tb.vhdl`
  - `psl/`
    - `README.rst`
    - `subdir001/`
      - `README.rst`
      - `run.sh`
      - ...

Then, we might add these to the docs, in place of the current 'Quick Start Guide' page. At the same time, it would be easier to use them as part of the test suites.

Hence, you can create a first PSL example in `examples/psl/<name>`. Nonetheless, I think this might be better handled as a separate PR.

Reply:
Sounds good. We could also link from other documentation sections to these examples for clarity. Will make another separate PR when I have some examples.